worlds/stardew_valley/logic/mine_logic.py

```diff
@@ -59,7 +58,8 @@ def can_progress_in_the_mines_from_floor(self, floor: int) -> StardewRule:
         weapon_rule = self.logic.mine.get_weapon_rule_for_floor_tier(tier)
         rules.append(weapon_rule)
 
-        if self.options.tool_progression & ToolProgression.option_progressive:
+        # No alternative for vanilla because we assume you will get ores to upgrade your tools in the mines.

```

Comment:
Is this true? What if you don't have access to Clint?

Reply:
Good question. ~~You probably should ask that to the person who wrote that rule.~~

Would adding something like `can_upgrade_tools` in tool logic make sense? That would check for Clint and money.

Reply:
The bug was there before? 😰

Reply:
Yeah, I just added a comment to explain why we only check tools when we're in progressive mode. I did not touch the rule itself